opennms-services/src/main/java/org/opennms/netmgt/snmpinterfacepoller/SnmpPoller.java

```diff
@@ -240,10 +242,21 @@ protected void scheduleNewSnmpInterface(String ipaddr) {
      * <p>scheduleExistingSnmpInterface</p>
      */
     protected void scheduleExistingSnmpInterface() {
-        
-    	for (OnmsIpInterface iface : getNetwork().getContext().getPollableNodes()) {
-            schedulePollableInterface(iface);    		
-    	}
+        ExecutorService executor = getExecutorService();
+        for (OnmsIpInterface iface : getNetwork().getContext().getPollableNodes()) {
+            executor.execute(() -> schedulePollableInterface(iface));
+        }
+        try {
+            if (executor.awaitTermination(60, TimeUnit.MINUTES)){
+                LOG.info("All snmp interfaces were scheduled successfully!");
```

Comment:
Don't executors only terminate if you terminate them, or shut down the JVM?  I suspect the poller `LegacyScheduler` will just keep on trucking until shutdown and this will never fire...

If you want to know if everything that was scheduled actually completed, you will probably have to wait on all its futures. I guess that kind of defeats the point of trying to do them asynchronously though, lol, unless you make a separate thread that watches for futures to complete and logs and/or handles errors.

Regardless of that, 60 minutes is a ridiculously long amount of time if these things are getting scheduled in parallel.

Reply:
Especially if using the default 5 minute interval.

Reply:
> Don't executors only terminate if you terminate them, or shut down the JVM? I suspect the poller `LegacyScheduler` will just keep on trucking until shutdown and this will never fire...
> 
> If you want to know if everything that was scheduled actually completed, you will probably have to wait on all its futures. I guess that kind of defeats the point of trying to do them asynchronously though, lol, unless you make a separate thread that watches for futures to complete and logs and/or handles errors.
> 
> Regardless of that, 60 minutes is a ridiculously long amount of time if these things are getting scheduled in parallel.

Yeah i forgot to add the .shutdown(), When testing a small local project it works when adding the .shutdown() before awaitTermination :)
What would be a good timeout value here? I just put a high value to not accidentally abort the scheduling if it takes along time.

Reply:
Right, but what I'm getting at is that I believe the scheduler injected into your poller is shared by many things, it comes from Pollerd. I think you would be shutting down the entire Pollerd scheduler as soon as init completes...

Reply:
I think instead you just need to keep track of the futures you made, and wait for them to finish. Don't try to control the scheduler, it's not `SnmpPoller`'s place to manipulate it.

(This really is exposing a bad API, the monitor instead should be asking Pollerd to schedule something instead of being handed a complete executor.)

Reply:
Aah! I see what you mean now. You're right, I will se what I can instead

Reply:
@RangerRick is this a good solution? 🙂